File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;
// 참여한 선수들의 이름 배열, 완주한 선수들의 이름 배열
string solution(vector<string> participant, vector<string> completion) {
    string answer = ""; // 완주하지 못한 선수의 이름 (한명)
    
    unordered_map<string, int> m;
    for(string part: participant){
        m[part]++;
    }
    
    for(string comp: completion){
        m[comp]--;
    }
    
    for(auto mm: m){
        if(mm.second != 0) answer = mm.first;
    }
    
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap; 
import java.util.List; 
import java.util.Map; 

class Solution {
    
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> map = new HashMap<>(); 
        for(String part : participant){
            map.put(part, map.getOrDefault(part, 0) + 1); 
        }
        
        for(String comp : completion){
            map.put(comp, map.get(comp)-1); 
        }
        
        for(Map.Entry<String, Integer> entry : map.entrySet()){
            if(entry.getValue() != 0){
                answer = entry.getKey(); 
                break; 
            }
        }
        
        return answer;
    }
}